components/test-page-setup.js

```diff
@@ -242,75 +292,81 @@ function VariantOptions(props) {
       </select>
     </>)
 }
-function random(paramsref){
-  const [params, setRandom] = useState(paramsref)
-  let array1 = ["round-robin", "chunking"];
-  let array2 = ["load-store", "store-load", "load-load", "store-store"];
-  let scratchMem = 4*params.stressLineSize*params.stressTargetLines;
-  let testMem = params.memStride * Math.pow(2,6);
-  let memStride_ = randomGenerator(1,128);
-  let maxWorkgroups =  randomGenerator(4,1024);
-  let minWorkgroups = maxWorkgroups+1;
-  while(minWorkgroups > maxWorkgroups){
-    minWorkgroups = randomGenerator(4,maxWorkgroups);;
-  }
-  let stressLineSize_ = randomGenerator(1,128);
-  let stressTargetLines_ =  randomGenerator(1,128);
-  scratchMem = 4*stressLineSize_ * stressTargetLines_;
-  while(testMem > 4096){
-    memStride_ = randomGenerator(1,128);
-    testMem = memStride_ * Math.pow(2,6);
-  }
-  setRandom({
-      minWorkgroups : minWorkgroups, 
-      maxWorkgroups : maxWorkgroups, 
-      testMemorySize : testMem,
-      memStride : memStride_,
-      memStressIterations : randomGenerater(0,1024),
-      preStressPct :randomGenerater(0,100),
-      preStressIterations :  randomGenerater(0,2048),
-      stressLineSize : stressLineSize_,
-      stressTargetLines : stressTargetLines_,
-      shufflePct : randomGenerater(0,100),
-      barrierPct : randomGenerater(0,100),
-      memStressPct : randomGenerater(0,100),
-      scratchMemorySize : scratchMem,
-      stressAssignmentStrategy : array1[Math.floor(Math.random() * 2)],
-      memStressPattern : array2[Math.floor(Math.random() * 4)],
-      preStressPattern : array2[Math.floor(Math.random() * 4)]
-  })
-  
-  console.log(params);
-}
-//need to be fixed
-var arrayObj = [];
-function doTuning(params ,numTuning){
-  for(let i = 0; i<=numTuning; i++){
-   random(params);
-    let obj = {
+
+let rows = [];
+let currentParam;
+let config;
+//run litmus test for each random config and store config for displaying 
+async function random(pageState, activeShader, testState,tuningTimes){
+ pageState.rows.update([]);
+ rows.splice(0,rows.length);
+ var keys;
+ pageState.running.update(true);
+ if (testState.numOutputs == 1) {
+   keys = ["seq", "weak"];
+ } else if (testState.numOutputs == 2) {
+   keys = ["seq0", "seq1", "interleaved", "weak"];
+ } else if (testState.numOutputs == 4) {
+   keys = ["seq", "interleaved", "weak"];
+ }
+ clearState(testState, keys);
+  for(let i =0; i<tuningTimes; i++){
+    let array1 = ["round-robin", "chunking"];
+    let array2 = ["load-store", "store-load", "load-load", "store-store"];
+    let maxWorkgroups =  randomGenerator(4,1024);
+    let minWorkgroups = randomGenerator(4, maxWorkgroups);
+    let stressLineSize = Math.pow(2, randomGenerator(1,10));
+    let stressTargetLines = randomGenerator(1,16);
+    let memStride = Math.pow(2, randomGenerator(1, 9));
+    let obj ={
```

Comment:
maybe not for this PR, but it probably makes sense to consolidate this random configuration with the random configuration on the explore page at some point